packages/nimble-components/src/table/specs/cell-state-when-scrolling-hld.md

```diff
@@ -0,0 +1,106 @@
+    -   Firefox: Rect looks to be correct for all of the selection variants (within the table, and crossing the table's bounds)
+-   Compare start/end points against a `Range` with the table fully selected (`tableRange = new Range(); tableRange.selectNode(table)`) via `range.compareBoundaryPoints(Range.START_TO_START, tableRange)` and `range.compareBoundaryPoints(Range.END_TO_END, tableRange)`
+    -   Chrome/Safari: Works when the text selection is fully within the table
+    -   Chrome: Works when the text selection is partially in the table, partially outside the table
+    -   Safari: Doesn't work when the text selection is partially in the table, partially outside the table. In that case, the `compareBoundaryPoints` results match what we'd get if the text selection was only before / only after the table (unusable).
+    -   Firefox: Errors out with a `WrongDocumentError` (i.e. one range is in a shadow root, and the other isn't / is in a different shadow root)
+-   (Chrome only) Look at `table.shadowRoot.getSelection()`. If non-null and `rangeCount > 0` we have selected text somewhere in the table. Simplest approach, but this API is non-standard / not in the other browsers / may be removed at any time in the future.
+
+**Note / Behavior limitation in Firefox**: You can only select a single cell's text. Once you cross into another cell/column the selection gets cleared. The selection also resets if you start text selection from outside the table, then move your cursor within the table viewport (to the rows/cells) to try to select more text there. There doesn't seem to be anything for us to do about this, it's probably related to the table's use of Shadow DOM.
+
+**Implementation Plan:**  
+Since the code required to detect table selection is problematic (`try/catch`) and complex, doesn't work fully in Safari, and is potentially fragile, we currently don't plan to try and detect table text selection. This means that if table text is selected and the user scrolls, text (different text) will be selected. We don't consider that to be a primary table interaction, and it shouldn't harm functionality (it's easy to resolve for the user by clicking off to deselect text). It's also the same behavior as the current `sl-grid`/`smart-table` used in SystemLink Enterprise.
+
+We can re-evaluate this in the future if and when the Text Selection APIs improve in the context of Shadow DOM.
+
+### Blur Focused Controls in Cells
+
+**Implementation Plan:**  
+Logic to find if the table contains an active/focused element within a cell:  
+Start with `document.activeElement`. If non-null, check if that element is a shadow root. If so, continue recursively with `shadowRoot.activeElement`. Stop when we reach a `nimble-table-cell` or `null`.
+
+If we have a focused cell:
+
+-   If the cell has an action menu (note: in this case the focused element will be a `nimble-menu-item`), find the associated `nimble-menu-button`. If open, set `open` to `false`.
```

Comment:
I think as written this is trying to make a generic implementation and that generic implementation will cover the action menu. I think we should just own the action menu within the table instead since the plugin api is not responsible for it.